### ISSUE TITLE
[import error of ESM]

### ISSUE THREAD

**Issue by icepy** [ceramicstudio/self.id#62]
# Description

the project i created with create-react-app.

why import Provider is undefined ?

```js
import { Provider } from "@self.id/react";

console.log('Provider', Provider)
```

Can't import the named export 'Provider' (imported as 'Provider') from default-exporting module (only default export is available)

```json
"@self.id/react": "^0.2.0",
"@self.id/web": "^0.3.0",
```

```js
import selfid from "@self.id/react";

console.log('selfid', selfid)
```

it's /static/media/lib.5cbed25f96df6e92ee98.cjs path. i didn't realdy understand. :( 

# Technical Information

Provide an explanation of the technical work to be done.

**Comment by utt6161** [comment]
Stumbled upon the same problem as well, but it doesn't seem like an issue with "@self.id/react" package, but rather with how CRA configures things.

Here's the repo with just "CRA --template typescript" and "@self.id/react" installed:
https://github.com/utt6161/cra-import-failure

Also, downgrading CRA doesn't solve the issue. Tried with react-scripts@4.0.0 and typescript@4.0.8 and i've got:
```console
./src/index.tsx
Attempted import error: 'Provider' is not exported from '@self.id/react'.
```

Downgrading anything further seems pointless.

**Comment by PaulLeCam** [comment]
CRA v4 won't work for sure, but `@self.id/react` should work with CRA v5.

**Comment by BenoistP** [comment]
Hi,

same issue :
- created a new app using :

1. npm uninstall -g create-react-app
2. npx create-react-app --template typescript cra-v5-ts
3. npm install '@self.id/framework'

It seems i didn't miss a step checking these resources :
https://betterprogramming.pub/an-in-depth-guide-for-create-react-app-5-cra-5-b94b03c233f2
https://create-react-app.dev/docs/adding-typescript/

But launch raises :

_ERROR in ./src/App.tsx 39:30-38
**Can't import the named export 'Provider' (imported as 'Provider') from default-exporting module (only default export is available)**     
 @ ./src/index.tsx 7:0-24 11:33-36_

Here's a repo of the step described above :

https://github.com/BenoistP/cra-v5-ts-self.id

Attempting update
-  npm install --save react-scripts@latest
does not change anything

**Comment by PaulLeCam** [comment]
Thanks, so this is really specific to CRA which we don't actively support, please use one of the [provided templates](https://github.com/ceramicstudio/self.id#templates) instead to setup a project.

**Comment by thiduzz** [comment]
@PaulLeCam - I don't think this problem is specific to CRA - it's for any ESM (including the vite example in the `templates-esm` folder which currently doesnt work).

Tried creating a Vite app from scratch as well and it did not work:

1. `yarn create vite auth-vite --template react-ts`
2. `cd auth-vite`
3. `yarn`
4. `yarn add @self.id/framework`
5. Adapt the main.tsx
```
import React from 'react'
import ReactDOM from 'react-dom'
import './index.css'
import App from './App'
import { Provider } from '@self.id/framework'
import closeIcon from '@self.id/multiauth/assets/icon-close.svg'
import selectedIcon from '@self.id/multiauth/assets/icon-selected.svg'
import ethereumLogo from '@self.id/multiauth/assets/ethereum.png'
import metaMaskLogo from '@self.id/multiauth/assets/metamask.png'

ReactDOM.render(
  <React.StrictMode>
    <Provider
      auth={{
        modal: { closeIcon, selectedIcon },
        networks: [
          {
            key: 'ethereum',
            logo: ethereumLogo,
            connectors: [{ key: 'injected', logo: metaMaskLogo }],
          },
        ],
      }}
      ui={{ full: true, style: { display: 'flex' } }}>
      <App />
    </Provider>
  </React.StrictMode>,
  document.getElementById('root')
)

```
6. `yarn dev`
7. Compiles normally but get the following console error: `Uncaught SyntaxError: The requested module '/node_modules/@self.id/framework/dist/lib.cjs?import' does not provide an export named 'Provider'`

Node version: v16.13.1
NPM version: 8.1.2
Yarn version: v1.22.4

**Comment by PaulLeCam** [comment]
@thiduzz this is expected, the current version of the SDK is exported as CommonJS only. ESM support will come in the next version.

**Comment by carlbarrdahl** [comment]
> @thiduzz this is expected, the current version of the SDK is exported as CommonJS only. ESM support will come in the next version.

Any updates on this @PaulLeCam ? When can we expect the next version? Is there something the community can help out with?

**Comment by thiduzz** [comment]
@carlbarrdahl - apparently this is already in the `develop` branch. I guess it will be out on the next release. Am i right @PaulLeCam ?

**Comment by PaulLeCam** [comment]
Yes It's already in develop and the packages are published on npm under the `next` tag. These versions depend on other Ceramic libraries that are also in RC and published under the `next` tag.

**Comment by carlbarrdahl** [comment]
Amazing! Thank you @PaulLeCam and @thiduzz
